crates/turbo-updater/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+use std::time::Duration;
+
+use colored::*;
+use serde::Deserialize;
+use thiserror::Error as ThisError;
+use update_informer::{Check, Package, Registry, Result as UpdateResult};
+
+mod ui;
+
+const DEFAULT_TIMEOUT: Duration = Duration::from_millis(800);
+const DEFAULT_INTERVAL: Duration = Duration::from_secs(60 * 60 * 24);
+
+#[derive(ThisError, Debug)]
+pub enum UpdateNotifierError {
+    #[error("Failed to write to terminal")]
+    RenderError(#[from] ui::utils::GetDisplayLengthError),
+}
+
+#[derive(Deserialize)]
+struct NpmVersionData {
+    version: String,
+}
+
+struct NPMRegistry;
+
+impl Registry for NPMRegistry {
+    const NAME: &'static str = "npm_registry";
+    fn get_latest_version(pkg: &Package, _timeout: Duration) -> UpdateResult<Option<String>> {
+        let url = format!("https://registry.npmjs.org/{}/latest", pkg);
+        let resp = ureq::get(&url).timeout(_timeout).call()?;
+        let result = resp.into_json::<NpmVersionData>().unwrap();
+        Ok(Some(result.version))
+    }
+}
+
+pub fn check_for_updates(
+    package_name: &str,
+    github_repo: &str,
+    footer: Option<&str>,
+    current_version: &str,
+    timeout: Option<Duration>,
+    interval: Option<Duration>,
+) -> Result<(), UpdateNotifierError> {
+    let timeout = timeout.unwrap_or(DEFAULT_TIMEOUT);
+    let interval = interval.unwrap_or(DEFAULT_INTERVAL);
+    let informer = update_informer::new(NPMRegistry, package_name, current_version)
+        .timeout(timeout)
+        .interval(interval);
+    if let Some(version) = informer.check_version().ok().flatten() {
```

Comment:
```suggestion
    if let Ok(Some(version)) = informer.check_version() {
```

I'd stick with this, currently it kinda "obfuscates" the error